ui/src/app/pages/workspace/workspace-edit.tsx

```diff
@@ -1048,6 +1054,29 @@ export const WorkspaceEdit = fp.flow(
       this.saveWorkspace();
     }
 
+    private async apiCreateWorkspaceAsync() {
+      let pollTimedOut = false;
+      setTimeout(() => (pollTimedOut = true), CREATE_WORKSPACE_TIMEOUT_MS);
+      let workspaceOp = await workspacesApi().createWorkspaceAsync(
+        this.state.workspace
+      );
+      while (
+        !pollTimedOut &&
+        workspaceOp.status === WorkspaceOperationStatus.PENDING
+      ) {
+        await new Promise((resolve) =>
+          setTimeout(resolve, CREATE_WORKSPACE_INTERVAL_MS)
+        );
+        workspaceOp = await workspacesApi().getWorkspaceOperation(
+          workspaceOp.id
+        );
+      }
+      if (workspaceOp.status === WorkspaceOperationStatus.ERROR) {
```

Comment:
```suggestion
      if (workspaceOp.status !== WorkspaceOperationStatus.SUCCESS) {
```
This way you would throw the error also in the case where the operation timed out.

Reply:
good catch, thanks